### DISCUSSION TITLE
Property 'message' does not exists on type 'WebSocket'

### DISCUSSION THREAD

**Discussion by brielov** [solidjs-community/solid-primitives#517] [Q&A]
Hi, I was following the example setup on `@solid-primitives/websocket` and the following line does not seem to work, at least for typescript.

```ts
const ws = createWS("ws://localhost:3000");
const state = createWSState(ws);

createEffect(on(ws.message, (msg) => console.log(msg), { defer: true }));
//                   ^ This does not work
```

Am I doing something wrong?

**Comment by atk** [comment]
Sorry, the documentation was not actualized. You have to use the message event to receive messages, as you can see in the [demo](createWSState).

**Reply by brielov** [reply]
Thanks for the reply. Although the link sends me to a 404 github discussion

**Reply by atk** [reply]
That is strange. It was supposed to point at the demo's src: https://github.com/solidjs-community/solid-primitives/blob/main/packages/websocket/dev/index.tsx
